grails-app/domain/org/pih/warehouse/invoice/InvoiceItemCandidate.groovy

```diff
@@ -84,6 +85,20 @@ class InvoiceItemCandidate {
         return unitPrice
     }
 
+    Product getAssociatedProduct() {
+        if (productCode) {
+            return Product.findByProductCode(productCode)
+        }
+        return null
+    }
+
+    Boolean isAdjustment() {
+        if (adjustmentAmount || adjustmentPercentage) {
+            return true
+        }
+        return false
+    }
```

Comment:
I am not sure, but I think it can be simplified to: `return adjustmentAmount || adjustmentPercentage`. Correct me if I'm wrong.

Reply:
it would then return not a Boolean, but the actual value of either `adjustmentAmount` or `adjustmentPercentage`

Reply:
If you want you can then cast this value to boolean using double bang operator 
```
return !!(adjustmentAmount || adjustmentPercentage)
```

Reply:
I know, but I also know that Justin is not a big fan of it, so I wanted to do it in a more common way

Reply:
Actually I think that since you have defined the function return type as _Boolean_ it should cast its return value implicitly.
I am not 100% sure about this but I think I have noticed this behavior before, so even if you would have
`return adjustmentAmount || adjustmentPercentage` 

By the way I am totally fine with how the function look right now and it is not a change request

Reply:
You guys were right - it is working fine - I expected it to work as it is in JS, so if you have something like:
`return "string" || null`, I'd expect it to return `"string"`, but it seems like Groovy is cleverer than I thought and if you write `||` or `&&` it parses to `Boolean` by itself, so in the case above I'd get `true` as the result, not `"string"`.
It's nice to know, as it's another Groovy's feature that works in a different way than in JS.

Reply:
As far as I know the following would definitely return a boolean
```
Boolean isAdjustment() { 
    return (adjustmentAmount || adjustmentPercentage)
}
```

Reply:
Oh sorry, just read @drodzewicz's comment. And yeah this would be the only way to go.